contracts/p1/BackingManager.sol

```diff
@@ -99,6 +99,84 @@ contract BackingManagerP1 is TradingP1, IBackingManager {
         }
     }
 
+    /// This will allow the callee to start a dust auction for a specific asset.
+    /// No other active trades must be open and current basket must be fully collateralized.
+    /// @param sellToken The asset we have a dust balance of
+    /// @custom:interaction RCEI
+    function startDustAuction(IERC20 sellToken, TradeKind kind) external notTradingPausedOrFrozen {
+        // @to reviewer: should we let the user set the TradeKind?
+
+        // == Refresh ==
+        // @to reviewer: Not sure what needs to be called/refreshed here
+        assetRegistry.refresh();
+        furnace.melt();
+
+        // == Checks/Effects ==
+
+        // checks: last like-kind auction must have ended + 1 block
+        uint48 previousAuctionEnd = tradeEnd[kind];
+        require(previousAuctionEnd + ONE_BLOCK < block.timestamp, "auction in progress");
+
+        // checks: we should not be doing anything important right now
+        require(tradesOpen == 0, "trade open");
+
+        // checks: basketHandler is OK and we're not trading too fast
+        require(basketHandler.isReady(), "basket not ready");
+        require(block.timestamp >= basketHandler.timestamp() + tradingDelay, "trading delayed");
+
+        // checks: Make sure we don't need to rebalance
+        BasketRange memory basketsHeld = basketHandler.basketsHeldBy(address(this));
+        require(basketsHeld.bottom >= rToken.basketsNeeded(), "undercollateralized");
+
+        // checks: We don't want to sell off basket tokens
+        // @to reviewer: I know that BasketHandletP1 can return just what we need
+        // So i'm not sure if this is the best way to do this otherwise,
+        // so we're quoting for 1 unit using .quote() and just using the returned ERC20s
+        (address[] memory erc20s, ) = basketHandler.quote(FIX_ONE, RoundingMode.CEIL);
+        require(!ArrayLib.contains(erc20s, address(sellToken)), "basket token");
+
+        (, uint192 rTokenPriceUSD) = basketHandler.price();
+        require(rTokenPriceUSD != 0, "cannot price rToken");
+
+        IAsset sellAsset = assetRegistry.toAsset(sellToken);
+
+        // checks: make sure we even hold a balance
+        uint192 sellQty = sellAsset.bal(address(this));
+        require(sellQty == 0, "zero balance");
+
+        // respect maxTradeVolume
+        uint192 maxTradeSize = sellAsset.maxTradeVolume();
+        sellQty = sellQty > maxTradeSize ? maxTradeSize : sellQty;
+
+        // checks: make sure all tokens can be priced
+        (uint192 sellTokenPriceUSD, ) = sellAsset.price();
+        require(sellTokenPriceUSD != 0, "cannot price token");
+
+        TradeRequest memory tradeRequest;
+        tradeRequest.sell = sellAsset;
+        tradeRequest.buy = assetRegistry.toAsset(IERC20(address(rToken)));
+        tradeRequest.sellAmount = sellQty;
+
+        // buyAmount = (sellQty * sellTokenPriceUSD) / rTokenPriceUSD
+        uint192 sellQtyInUSD = sellQty.mul(sellTokenPriceUSD, RoundingMode.CEIL);
```

Comment:
FYI you can just use `CEIL` here, it's an abbreviated version of `RoundingMode.CEIL`